File: web/src/types/library.ts
import { z } from 'zod/v4';

export const LibrarySchema = z.object({
  id: z.int().optional(),
  path: z.string({ error: '请输入路径' }),
  gallery_count: z.int().default(0).optional(),
  image_count: z.int().default(0).optional(),
  video_count: z.int().default(0).optional(),
  created_at: z.string().optional(),
  updated_at: z.string().optional(),
});

export type Library = z.infer<typeof LibrarySchema>;

export type LibraryCreate = Pick<Library, 'path'>;

export type LibraryUpdate = Pick<Library, 'path' | 'id'>;

File: dto/library.go
package dto

import "github.com/dromara/carbon/v2"

type Library struct {
	ID           int64           `json:"id"`
	Path         string          `json:"path"`
	GalleryCount int             `json:"gallery_count"`
	ImageCount   int             `json:"image_count"`
	VideoCount   int             `json:"video_count"`
	CreatedAt    carbon.DateTime `json:"created_at"`
	UpdatedAt    carbon.DateTime `json:"updated_at"`
}

type LibraryCreate struct {
	Path string `json:"path"`
}

type LibraryUpdate struct {
	ID   int64  `json:"id"`
	Path string `json:"path"`
}

type LibraryDelete struct {
	IDs []int64 `json:"ids"`
}
